code/controllers/stonedmc/subsystem/weeds.dm

```diff
@@ -0,0 +1,70 @@
+SUBSYSTEM_DEF(weeds)
+	name = "Weed"
+	priority = FIRE_PRIORITY_WEED
+	runlevels = RUNLEVEL_GAME
+	wait = 5 SECONDS
+
+	// This is a list of nodes on the map.
+	var/list/weed_nodes = list()
+	var/list/currentrun
+
+/datum/controller/subsystem/weeds/Initialize(start_timeofday)
+	return ..()
+
+/datum/controller/subsystem/weeds/stat_entry()
+	return ..("Nodes: [length(weed_nodes)]")
+
+/datum/controller/subsystem/weeds/fire(resumed = FALSE)
+	if(!resumed)
+		currentrun = deepCopyList(weed_nodes)
+
+	var/obj/effect/alien/weeds/node/N
+	while(length(currentrun))
+		N = currentrun[currentrun.len]
+		currentrun.len--
+		if(!N)
+			weed_nodes.Remove(N)
+			continue
+
+		// PROCESS N
+		for(var/turf/T in N.node_turfs) 
```

Comment:
If you don't expect the list to have anything but turfs, it's more efficient to do
```
for(var/X in N.node_turfs) 
	var/turf/T = X
```
As it saves an implicit `istype()` check.